src/bp/nlu-core/predict-pipeline.ts

```diff
@@ -199,18 +187,22 @@ async function predictContext(input: PredictStep, predictors: Predictors): Promi
   }
 }
 
-async function predictIntent(input: PredictStep, predictors: Predictors): Promise<PredictStep> {
-  if (predictors.intents.length === 0) {
-    return { ...input, intent_predictions: { per_ctx: { [DEFAULT_CTX]: [{ label: NONE_INTENT, confidence: 1 }] } } }
+async function predictIntent(input: ContextStep, predictors: Predictors): Promise<IntentStep> {
+  if (_.flatMap(predictors.intents, i => i.utterances).length <= 0) {
+    return {
+      ...input,
+      intent_predictions: { per_ctx: { [DEFAULT_CTX]: [{ label: NONE_INTENT, confidence: 1 }] } }
+    }
   }
 
   const customEntities = getCustomEntitiesNames(predictors)
-  const ctxToPredict = input.ctx_predictions!.map(p => p.label)
+
+  const ctxToPredict = input.ctx_predictions.map(p => p.label)
   const predictions = (
     await Promise.map(ctxToPredict, async ctx => {
       let preds: sdk.MLToolkit.SVM.Prediction[] = []
 
-      const predictor = predictors.intent_classifier_per_ctx![ctx]
+      const predictor = predictors.intent_classifier_per_ctx?.[ctx]
```

Comment:
in some cases, when there was an intent with no utterance, `predictors.intent_classifier_per_ctx` was undefined.